pkg/gui/style/text_style.go

```diff
@@ -0,0 +1,160 @@
+package style
+
+import (
+	"github.com/gookit/color"
+)
+
+// A TextStyle contains a foreground color, background color, and
```

Comment:
I ran some local benchmarks to compare this code with:
1) the original PR
2) the master branch
3) this PR but calling deriveStyle every time we call `SPrintf`.

I loaded 30k commits and tested how long it took to generate the whole list in `pkg/gui/presentation/commits.go`. The results were:

```
master: 50ms roughly
original PR's branch: around 30-40ms
my inefficient branch: 35-45ms
my efficient branch: 29-38ms
```
it's only really a 10ms gap and for loading 30k commits I think that's pretty good. I'm tempted to switch to the inefficient approach for simplicity but we may come across situations where the gap becomes bigger